controllers/kmsissuer_controller.go

```diff
@@ -98,7 +104,72 @@ func (r *KMSIssuerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			return ctrl.Result{}, r.manageFailure(ctx, log, issuer, err, "Failed to update the issuer with the issued Certificate")
 		}
 	}
-	return ctrl.Result{}, r.manageSuccess(ctx, log, issuer)
+	return ctrl.Result{
+		RequeueAfter: time.Until(desiredCert.NotAfter.Add(-1 * issuer.Spec.RenewBefore.Duration)),
+	}, r.manageSuccess(ctx, log, issuer)
+}
+
+// setIssuerDefaultValues
+func setIssuerDefaultValues(issuer *kmsiapi.KMSIssuer) {
+	if issuer.Spec.Duration == nil || issuer.Spec.Duration.Duration == 0 {
+		issuer.Spec.Duration = &metav1.Duration{Duration: DefaultCertDuration}
+	}
+	if issuer.Spec.RenewBefore == nil {
+		issuer.Spec.RenewBefore = &metav1.Duration{
+			Duration: issuer.Spec.Duration.Duration * DefaultCertRenewalRatio,
+		}
+	}
+	if issuer.Spec.RenewBefore.Duration > issuer.Spec.Duration.Duration {
```

Comment:
Is there a way to throw a validation error instead of forcing defaults?

Reply:
Same as above. I am trying to follow the same convention that cert-manager follows

https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1.CertificateSpec